commit c2cc1adb60c759678ed872c943badeddb1977e21
fix: maximum diversification with negative expected returns (#197)


diff --git a/src/skfolio/optimization/convex/_mean_risk.py b/src/skfolio/optimization/convex/_mean_risk.py
index b5858df..12eae33 100644
--- a/src/skfolio/optimization/convex/_mean_risk.py
+++ b/src/skfolio/optimization/convex/_mean_risk.py
@@ -1079,14 +1079,19 @@ class MeanRisk(ConvexOptimization):
                 )
             case ObjectiveFunction.MAXIMIZE_RATIO:
                 # Capture common obvious mistake before solver failure to help user
-                if np.isscalar(self.min_weights) and self.min_weights >= 0:
-                    if np.max(return_distribution.mu) - self.risk_free_rate <= 0:
-                        raise ValueError(
-                            "Cannot optimize for Maximum Ratio with your current "
-                            "constraints and input. This is because your assets' "
-                            "expected returns are all under-performing your risk-free "
-                            f"rate {self.risk_free_rate:.2%}."
-                        )
+                if (
+                    self.overwrite_expected_return is None
+                    and np.isscalar(self.min_weights)
+                    and self.min_weights >= 0
+                    and np.max(return_distribution.mu) - self.risk_free_rate <= 0
+                ):
+                    raise ValueError(
+                        "Cannot optimize for Maximum Ratio with your current "
+                        "constraints and input. This is because your assets' "
+                        "expected returns are all under-performing your risk-free "
+                        f"rate {self.risk_free_rate:.2%}."
+                    )
+
                 homogenization_factor = _optimal_homogenization_factor(
                     mu=return_distribution.mu
                 )
